src/import/pull-tar.c

```diff
@@ -315,11 +315,13 @@ static void tar_pull_job_on_finished(PullJob *j) {
         if (!tar_pull_is_done(i))
                 return;
 
-        if (i->checksum_job->style == VERIFICATION_PER_DIRECTORY && i->signature_job->error != 0) {
-                log_error_errno(j->error, "Failed to retrieve signature file, cannot verify. (Try --verify=no?)");
+        if (i->signature_job) {
+                if (i->checksum_job->style == VERIFICATION_PER_DIRECTORY && i->signature_job->error != 0) {
```

Comment:
hmm, not sure i follow. how can this even happen? if style is VERIFICATION_PER_DIRECTORY, then we really should have a signature job, no?